concent_api/core/exceptions.py

```diff
@@ -1,7 +1,18 @@
+from utils.constants import ErrorCode
+
 
 class UnexpectedResponse(Exception):
     pass
 
 
 class Http400(Exception):
-    pass
+    error_code = None
```

Comment:
I have some doubts about `__init__` of this exception. The way it is constructed enforces that, `error_message` must be provided as positional argument and `error_code` must be provided as keyword argument. This seems to overcomplicate things...
I would propse to change the method to sth like this:
```
    def __init__(self, error_message, error_code):
        super().__init__(error_message)
        assert isinstance(error_code, ErrorCode)
        self.error_code = error_code
```
It simplifies constructor and eliminates the need of class fields: `error_code=None`, and `error_message=''`

Reply:
In general it's a good point but for this particular exception I think that mandatory error code is a good idea. I can't think of a situation when we would want to return it without a code. So I'm for leaving it as is, without a default.

Reply:
Sorry, my comment was not clear enough. I also commented in a hurry just before the meeting and did not look at the code close enough.

Arguments are indeed mandatory and @dybi's suggestion preserves that so please ignore that remark.

Upon looking at it closely I completely agree with @dybi. The constructor would be better and simpler having normal, named arguments rather than relying on `args` and `kwargs`.

Reply:
One note - `error_message` must be set as an instance attribute, otherwise there is no other way to get pure message in decorators.

Reply:
:+1: done